src/Nancy/HttpLink.cs

```diff
@@ -0,0 +1,257 @@
+namespace Nancy
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Text;
+    using Nancy.Responses.Negotiation;
+
+    /// <summary>
+    /// Represents one of possibly many RFC 5899 HTTP Links contained in a <see cref="HttpLinkHeader"/>.
+    /// </summary>
+    public class HttpLink : Dictionary<string, object>, IEquatable<HttpLink>
+    {
+        private readonly Uri targetUri;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="HttpLink" /> class.
+        /// </summary>
+        /// <param name="targetUri">The target URI of the link.</param>
+        /// <param name="relation">The relation that identifies the semantics of the link.</param>
+        /// <param name="type">The optional type parameter is a hint indicating what the media type of the result of dereferencing the link should be. Note that this is only a hint; for example, it does not override the HTTP Content-Type header of a HTTP response obtained by actually following the link.</param>
+        /// <param name="title">The optional title parameter is used to label the destination of a link such that it can be used as a human-readable identifier (e.g., a menu entry) in the language indicated by the HTTP Content-Language header (if present).</param>
+        /// <exception cref="System.ArgumentNullException"><paramref name="targetUri" /> or <paramref name="relation" /></exception>
+        public HttpLink(string targetUri, string relation, string type = null, string title = null)

```

Comment:
@khellang Yea, that's kind of why I had the load of constructors @thecodejunkie complained about. :wink: But by reducing the number of types to basically just allow `string` for most parameters, I've managed to get the number of constructors down to 4. I hope that's a fine number. :smile: